core/src/main/java/org/apache/druid/java/util/common/DateTimes.java

```diff
@@ -107,7 +107,12 @@ public static DateTime utc(long instant)
 
   public static DateTime of(String instant)
   {
-    return new DateTime(instant, ISOChronology.getInstanceUTC());
+    try {
+      return new DateTime(instant, ISOChronology.getInstanceUTC());
+    }
+    catch (IllegalArgumentException ex) {
+      return new DateTime(Long.valueOf(instant), ISOChronology.getInstanceUTC());
```

Comment:
This will obscure the original exception, by reporting something like "java.lang.NumberFormatException: For input string" rather than the original DateTime String parsing exception. How about doing this instead:

```java
    try {
      return new DateTime(instant, ISOChronology.getInstanceUTC());
    }
    catch (IllegalArgumentException ex) {
      try {
        return new DateTime(Long.valueOf(instant), ISOChronology.getInstanceUTC());
      } catch (IllegalArgumentException ex2) {
        // Throw the original exception, since it is more likely to be meaningful to the user.
        throw ex;
      }
    }
```

Reply:
An oversight on my part. I'll make the change. Thanks